server.go

```diff
@@ -418,6 +418,35 @@ func (server *Server) listenSignals() {
 	server.Stop()
 }
 
+func createClientTLSConfig(tlsOption *TLS) (*tls.Config, error) {
+	if tlsOption == nil {
+		return nil, errors.New("no TLS provided")
```

Comment:
since it's a pointer, a tlsOption(nil) will panic. It should really be checked prior to access